docs/docusaurus/docs/cloud/alerts/_examples/create_a_custom_action.py

```diff
@@ -0,0 +1,91 @@
+"""
+This is an example script for how to create a custom Action.
+
+To test, run:
+pytest --docs-tests -k "docs_example_create_a_custom_action" tests/integration/test_script_runner.py
+"""
+
+# EXAMPLE SCRIPT STARTS HERE:
+
+# <snippet name="docs/docusaurus/docs/cloud/alerts/_examples/create_a_custom_action.py - full code example">
+
+from typing import Literal, Union
+
+from typing_extensions import override
+
+from great_expectations.checkpoint import (
+    ActionContext,
+    CheckpointResult,
+    ValidationAction,
+)
+
+
+# 1. Extend the `ValidationAction` class.
+# <snippet name="docs/docusaurus/docs/cloud/alerts/_examples/create_a_custom_action.py - extend class">
+class MyCustomAction(ValidationAction):
+    # </snippet>
+
+    # 2. Set the `type` attribute to a unique string that identifies the Action.
+    # <snippet name="docs/docusaurus/docs/cloud/alerts/_examples/create_a_custom_action.py - set type">
+    type: Literal["my_custom_action"] = "my_custom_action"
+    # </snippet>
+
+    # 3. Optional. Add any additional fields your Action requires at runtime.
+    # <snippet name="docs/docusaurus/docs/cloud/alerts/_examples/create_a_custom_action.py - add custom fields">
+    my_custom_str_field: str
+    jira_api_url: str
+    # </snippet>
+
+    # 4. Override the `run()` method to perform the desired task.
+    # <snippet name="docs/docusaurus/docs/cloud/alerts/_examples/create_a_custom_action.py - override run">
+    @override
+    def run(
+        self,
+        checkpoint_result: CheckpointResult,
+        action_context: Union[
+            ActionContext, None
+        ],  # Contains results from prior Actions in the same Checkpoint run.
+    ) -> dict:
+        # Domain-specific logic
+        self._do_my_custom_action(checkpoint_result)
+        # Optional. Access custom fields you provide the Action at runtime.
+        extra_context = self.my_custom_str_field
+        jira_api_url = self.jira_api_url
+        # Return information about the Action
+        return {
+            "some": "info",
+            "extra_context": extra_context,
+            "jira_api_url": jira_api_url,
+        }
+
+    def _do_my_custom_action(self, checkpoint_result: CheckpointResult):
+        # Perform custom logic based on the validation results.
+        #
+        # from atlassian import Jira
+        # from requests import HTTPError
+
+        # jira = Jira(
+        #     url   = self.jira_api_url,
+        #     token = "MyPersonalAccessToken"
```

Comment:
While this is commented out as a theoretical example, I suspect that opening an issue in Jira is a common use case so it would be ideal if folks could uncomment this and have it work. 

I was a little confused by the placeholder `MyPersonalAccessToken` because I first thought this referred to my GX Cloud user access token. Then when I was testing the code I found that this syntax didn't work and I needed to pass credentials as a username and password. Did you get this `token` stynax to work? Here's a suggestion for the syntax that worked for me

 ```suggestion
        #     username = "my_jira_user@company.com",
        #     password = "my_jira_API_token"
```

Reply:
I realized I was using an older Atlassian package, so I've updated the code. Generally we should use examples that don't use username and password authentication, so this updated version should work. Let me know if it doesn't.

Reply:
This updated code does not work for me. Details in a new comment thread at https://github.com/great-expectations/great_expectations/pull/11521#discussion_r2547917793